clock/clock.go

```diff
@@ -95,3 +95,13 @@ func (r *realTimer) Stop() bool {
 func (r *realTimer) Reset(d time.Duration) bool {
 	return r.timer.Reset(d)
 }
+
+// SinceInMicroseconds gets the time since the specified start in microseconds.
+func SinceInMicroseconds(start time.Time) float64 {
```

Comment:
This seems to be missing the point of this file. 
This file exists to give us a mechanism to allow us to inject a fake clock.
That fake clock can then be used to control the flow of time in tests to prevent tests from being flaky.
 
I have no issue with having SinceInUnit() methods but they should either be methods of the clock interface (so they can be overriden with the fake clock) or they should take the clock interface as a parameter (so that then call since on with that parameter and hence get the fake implementation when desired).

Some Examples of this include 
https://github.com/kubernetes/apiserver/blob/master/pkg/storage/cacher/watch_cache.go#L305
https://github.com/kubernetes/apiserver/blob/master/pkg/storage/cacher/watch_cache_test.go#L81